ResearchKit/Common/ORKReviewStepViewController.m

```diff
@@ -0,0 +1,213 @@
+            BOOL includeStep = [obj isKindOfClass:[ORKQuestionStep class]] || [obj isKindOfClass:[ORKFormStep class]] || [obj isKindOfClass:[ORKInstructionStep class]];
+            if (includeStep) {
+                [filteredSteps addObject:obj];
+            }
+        }];
+        _steps = [filteredSteps copy];
+        _resultSource = [self reviewStep].resultSource != nil ? [self reviewStep].resultSource : resultSource;
+    }
+    return self;
+}
+
+- (void)viewWillAppear:(BOOL)animated {
+    [super viewWillAppear:animated];
+    [self.taskViewController setRegisteredScrollView: _tableContainer.tableView];
+}
+
+- (void)viewDidAppear:(BOOL)animated {
+    [super viewDidAppear:animated];
+    UIAccessibilityPostNotification(UIAccessibilityScreenChangedNotification, self.navigationItem.leftBarButtonItem);
+}
+
+- (void)setContinueButtonItem:(UIBarButtonItem *)continueButtonItem {
+    [super setContinueButtonItem:continueButtonItem];
+    _continueSkipView.continueButtonItem = continueButtonItem;
+}
+
+- (void)setLearnMoreButtonItem:(UIBarButtonItem *)learnMoreButtonItem {
+    [super setLearnMoreButtonItem:learnMoreButtonItem];
+    _tableContainer.stepHeaderView.learnMoreButtonItem = self.learnMoreButtonItem;
+}
+
+- (void)setSkipButtonItem:(UIBarButtonItem *)skipButtonItem {
+    [super setSkipButtonItem:skipButtonItem];
+    _continueSkipView.skipButtonItem = self.skipButtonItem;
+}
+
+- (void)stepDidChange {
+    [super stepDidChange];
+    
+    [_tableContainer removeFromSuperview];
+    _tableContainer = nil;
+    
+    _tableContainer.tableView.delegate = nil;
+    _tableContainer.tableView.dataSource = nil;
+    _continueSkipView = nil;
+    
+    if ([self reviewStep]) {
+        _tableContainer = [[ORKTableContainerView alloc] initWithFrame:self.view.bounds];
+        _tableContainer.tableView.delegate = self;
+        _tableContainer.tableView.dataSource = self;
+        _tableContainer.tableView.clipsToBounds = YES;
+
+        [self.view addSubview:_tableContainer];
+        _tableContainer.tapOffView = self.view;
+        
+        _tableContainer.stepHeaderView.captionLabel.useSurveyMode = self.step.useSurveyMode;
+        _tableContainer.stepHeaderView.captionLabel.text = [self reviewStep].title;
+        _tableContainer.stepHeaderView.instructionLabel.text = [self reviewStep].text;
+        _tableContainer.stepHeaderView.learnMoreButtonItem = self.learnMoreButtonItem;
+        
+        _continueSkipView = _tableContainer.continueSkipContainerView;
+        _continueSkipView.skipButtonItem = self.skipButtonItem;
+        _continueSkipView.continueEnabled = YES;
+        _continueSkipView.continueButtonItem = self.continueButtonItem;
+        _continueSkipView.optional = self.step.optional;
+        [_tableContainer setNeedsLayout];
+    }
+}
+
+- (ORKReviewStep *)reviewStep {
+    return [self.step isKindOfClass:[ORKReviewStep class]] ? (ORKReviewStep *) self.step : nil;
+}
+
+//TODO: state restoration
```

Comment:
Arrgh... I got a question regarding `UIStateRestoring`conformance. `ORKTaskViewController`uses state restoration when being initialized from `restorationData`. But during that init `viewControllerForStep:` gets called, so why does the step view controller conforms to `UIStateRestoring` anyway?